File: domain/src/main/java/com/bitunified/ledconfig/productconfiguration/ModelChosenStep.java
package com.bitunified.ledconfig.productconfiguration;


import com.bitunified.ledconfig.domain.Model;
import com.bitunified.ledconfig.productconfiguration.steps.Step;


public class ModelChosenStep {


    private Step step;
    private Model chosenModel;
    private int modelValue;
    private boolean skipped;

    public int getModelValue() {
        return modelValue;
    }

    public void setModelValue(int modelValue) {
        this.modelValue = modelValue;
    }

    public Model getChosenModel() {
        return chosenModel;
    }

    public void setChosenModel(Model chosenModel) {
        this.chosenModel = chosenModel;
    }

    public Step getStep() {
        return step;
    }

    public void setStep(Step step) {
        this.step = step;
    }

    public boolean isSkipped() {
        return skipped;
    }

    public void setSkipped(boolean skipped) {
        this.skipped = skipped;
    }
}

File: web-steps/led-config/src/app/domain/ModelChosenStep.ts
import {StepModel} from "./StepModel";
import {Model} from "./Model";
import { serialize,serializeAs,deserialize,deserializeAs } from 'cerialize';
export class ModelChosenStep {

  @serializeAs(StepModel)
  @deserializeAs(StepModel)
  public step:StepModel;

  @serializeAs(Model)
  @deserializeAs(StepModel)
  chosenModel:Model;

  @serialize
  @deserialize
  modelValue:number;

  @serialize
    @deserialize
  skipped:boolean;
}
